File: lib/ts/src/internal/types/TError.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { TUnion } from './TUnion.js';

export class TError {
    name: string;
    errors: TUnion;

    constructor(name: string, errors: TUnion) {
        this.name = name;
        this.errors = errors;
    }
}

File: lib/go/internal/types/TError.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

// TError represents an error type wrapping an underlying union.
type TError struct {
	Name   string
	Errors *TUnion
}

// NewTError constructs a TError instance.
func NewTError(name string, errors *TUnion) *TError {
	return &TError{Name: name, Errors: errors}
}
